src/sagas/wallet.js

```diff
@@ -120,9 +126,12 @@ export function* startWallet(action) {
   yield put(setUseWalletService(useWalletService));
   yield put(setAvailablePushNotification(usePushNotification));
 
-  // We don't want to clean access data since as if something goes
-  // wrong here, the stored words would be lost forever.
-  walletUtil.cleanLoadedData({ cleanAccessData: false });
+  // clean storage and metadata before starting the wallet
+  // this should be cleaned when stopping the wallet,
+  // but the wallet may be closed unexpectedly
+  const storage = STORE.getStorage();
+  yield storage.store.cleanMetadata();
+  yield storage.cleanStorage(true);
```

Comment:
Are we keeping the access data as before?

Reply:
Yes, `cleanMetadata` and `cleanStorage` only affect tokens, history, utxos and possibly loaded addresses (which would require `storage.cleanStorage(true, true)`).

This means we are persisting access data and addresses.
The addresses should not be an issue unless we want to change the network, which is not currently possible.